st-exerciseviewer/src/main/java/de/saring/exerciseviewer/parser/impl/TimexPwxParser.java

```diff
@@ -691,14 +697,18 @@ private EVExercise parseWorkoutSamples(EVExercise exercise, Node workoutNode) {
                 exercise.getSampleList().add(sample);
 
                 // update Zone information
-                if (exercise.getHeartRateLimits() != null) {
+                if ((exercise.getHeartRateLimits() != null) && (! exercise.getHeartRateLimits().isEmpty())) {
+
                     for (int j = 0; j < 6; j++) {
-                        if (sample.getHeartRate() > exercise.getHeartRateLimits().get(j).getUpperHeartRate()) {
```

Comment:
Doesn't it make more sense to check whether exercise.getHeartrateLimits() cotaints an element at index j instead of just isEmpty()?

Reply:
I do not have enough context on the file format to answer this question. Why are we looking at the first six samples only? What happens if we skip a few?

My changes were  admittedly fairly mechanical, just to see if I can get this to parse (after the success with the fit parser).